src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const stringPart = this[i];
+
+      if (stringPart !== null && stringPart !== undefined) {
+        string += stringPart;
+      }
+
+      if (i < this.length - 1) {
```

Comment:
I suggest putting this.length in a variable, since we already use it twice, but I guess it's optional